server/test/repository/QuestionRepositoryTest.java

```diff
@@ -299,4 +300,159 @@ public void createOrUpdateDraft_managesUniversalTagCorrectly()
     question = repo.createOrUpdateDraft(nextQuestionDefinition);
     assertThat(question.getQuestionTags().contains(QuestionTag.UNIVERSAL)).isFalse();
   }
+
+  private QuestionDefinition addTagToDefinition(QuestionModel question)
```

Comment:
suggestion (non-blocking): These helper methods look great! I suggest moving them to the bottom of the file so all the tests are grouped together for readability.